public/app/features/explore/QueryLibrary/QueryTemplatesList.tsx

```diff
@@ -1,16 +1,70 @@
-import React from 'react';
+import { uniq } from 'lodash';
+import React, { useEffect, useState } from 'react';
 
+import { getBackendSrv } from '@grafana/runtime/src/services/backendSrv';
 import { EmptyState, Spinner } from '@grafana/ui';
 import { useAllQueryTemplatesQuery } from 'app/features/query-library';
+import { User } from 'app/features/query-library/api/types';
 import { QueryTemplate } from 'app/features/query-library/types';
 
 import { getDatasourceSrv } from '../../plugins/datasource_srv';
 
 import QueryTemplatesTable from './QueryTemplatesTable';
 import { QueryTemplateRow } from './QueryTemplatesTable/types';
 
+const getQueryTemplateRows = async (data: QueryTemplate[]): Promise<QueryTemplateRow[]> => {
+  const userQtList = uniq(data.map((qt) => qt.user).filter((user): user is User => !!user));
+
+  const userData = Promise.all(
+    userQtList.map((user) => {
+      if (user.userId) {
+        return getBackendSrv().get(`api/users/${user.userId}`);
```

Comment:
Apparently this may fail due to lack of permissions to read users :O I've created a new user and got this: 

<img width="1698" alt="Screenshot 2024-06-14 at 13 51 46" src="https://github.com/grafana/grafana/assets/745532/4cf1eb2b-351b-40e3-a3a2-e764b0ab0ada">

Not showing the user name is probably fine in such case but I think we still should show query templates 🤔

Reply:
Removed other call - no longer relevant